### PR TITLE
Required min max

### PR DESCRIPTION
**Pull request by lasalvavida**
Fills in accessor min and max if they aren't present as those will be required properties in glTF 1.0.1

### CODE REVIEW

**Comment by lasalvavida** [comment]
@lilleyse, this is ready for a look when you get a chance

File: lib/addDefaults.js [hunk 1]
```diff
@@ -19,8 +20,15 @@ function accessorDefaults(gltf) {
...
             accessor.byteStride = defaultValue(accessor.byteStride, 0);
+            if (!defined(accessor.min) || !defined(accessor.max)) {
+                var minMax = findAccessorMinMax(gltf, accessor);
+                accessor.min = defaultValue(accessor.min, minMax.min);
+                accessor.max = defaultValue(accessor.max, minMax.max);
+            }
         }
     }
+    
+
```
Comment (lilleyse) [inline lib/addDefaults.js]: Remove these two extra lines.

File: lib/findAccessorMinMax.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+            for (var i = 0; i < count; i++) {
+                for (var j = 0; j < numberOfComponents; j++) {
+                    var value = readBufferComponentType(source, componentType, byteOffset + j * componentByteLength);
+                    min[j] = Math.min(min[j], value);
+                    max[j] = Math.max(max[j], value);
+                }
+                byteOffset += byteStride;
+            }
+
```
Comment (lilleyse) [inline lib/findAccessorMinMax.js]: Extra line.

File: specs/lib/findAccessorMinMaxSpec.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        var gltfAccessor = gltf.accessors.accessor;
+        gltfAccessor.byteStride = 0;
+
+        var minMax = findAccessorMinMax(gltf, gltfAccessor);
+        var min = minMax.min;
+        var max = minMax.max;
+        for (var i = 0; i < min.length; i++) {
+            expect(min[i]).toEqual(expectMin[i]);
+            expect(max[i]).toEqual(expectMax[i]);
+        }
```
Comment (lilleyse) [inline specs/lib/findAccessorMinMaxSpec.js]: `expect(min).toEqual(expectMin);` should work as well.

**Comment by lilleyse** [comment]
Just those minor comments, this looks good.

**Comment by lasalvavida** [comment]
@lilleyse Updated with changes

**Comment by pjcozzi** [comment]
Part of https://github.com/KhronosGroup/glTF/issues/605.

File: lib/addDefaults.js [hunk 4]
```diff
@@ -19,6 +20,11 @@ function accessorDefaults(gltf) {
         if (accessors.hasOwnProperty(name)) {
             var accessor = accessors[name];
             accessor.byteStride = defaultValue(accessor.byteStride, 0);
+            if (!defined(accessor.min) || !defined(accessor.max)) {
```
Comment (pjcozzi) [inline lib/addDefaults.js]: Soon, we will separate this into a new stage that adds min/max defaults.  We will call it here and also from a new stage that converts 1.0 models to 1.0.1.

We also want to separate `addDefaults` into `addDefaults` and `convert 0.8 to `1.0`.  Right now it does both at the same time.

See #1 and https://github.com/KhronosGroup/glTF/wiki/glTF-0.8-to-1.0-Guide.

You are welcome to take a look at this now if you want.

**Pull request merged**